src/compile/mark/valueref.ts

```diff
@@ -373,7 +372,16 @@ export function midPoint({
     // In such case, we will use default ref.
   }
 
-  return isFunction(defaultRef) ? defaultRef() : defaultRef;
+  const ref = isFunction(defaultRef) ? {...defaultRef(), ...(offset ? {offset} : {})} : defaultRef;
+
+  if (ref) {
+    // for non-position, ref could be undefined.
+    return {
+      ...ref,
+      ...(offset ? {offset} : {})
```

Comment:
What if the offset is 0? 

Should this be:

```suggestion
      ...(offset != null ? {offset} : {})
```

Reply:
If offset is 0, you can safely omit the offset.

Reply:
Good. Just wanted to make sure. Maybe it's worth a comment here since `x ? .. : ...` always makes my alarm bells go off.